File: SBOLCanvasFrontend/src/app/variableComponentInfo.ts
import { CombinatorialInfo } from './combinatorialInfo';
import { GlyphInfo } from './glyphInfo';
import { IdentifiedInfo } from './identifiedInfo';

export class VariableComponentInfo {
    uri: string;
    variable: string;
    operator: string;
    variants: IdentifiedInfo[] = [];

    constructor(variable: string){
        this.variable = variable;
    }

    makeCopy(): VariableComponentInfo {
        const copy: VariableComponentInfo = new VariableComponentInfo(this.variable);
        copy.uri = this.uri;
        copy.operator = this.operator;
        for(let variant of this.variants){
            copy.variants.push(variant.makeCopy());
        }
        return copy;
    }

    addVariant(info: IdentifiedInfo){
        // avoid duplicates
        for(let variant of this.variants){
            if(variant.uri == info.uri)
                return;
        }
        this.variants.push(info);
    }
}

File: SBOLCanvasBackend/src/data/VariableComponentInfo.java
package data;

public class VariableComponentInfo {

	private String uri;
	private String variable;
	private String operator;
	private IdentifiedInfo[] variants;

	public String getUri() {
		return uri;
	}

	public void setUri(String uri) {
		this.uri = uri;
	}

	public String getVariable() {
		return variable;
	}

	public void setVariable(String variable) {
		this.variable = variable;
	}

	public String getOperator() {
		return operator;
	}

	public void setOperator(String operator) {
		this.operator = operator;
	}

	public IdentifiedInfo[] getVariants() {
		return variants;
	}

	public void setVariants(IdentifiedInfo[] variants) {
		this.variants = variants;
	}

}
